src/System.Management.Automation/engine/runtime/Operations/MiscOps.cs

```diff
@@ -3694,6 +3694,10 @@ private static string ArgumentToString(object arg)
 
         internal static void LogMemberInvocation(string targetName, string name, object[] args)
         {
+#if UNIX && !DEBUG
```

Comment:
Thanks for tackling this, but, given that there seems to be only one call to this method (in `Binders.cs`), wouldn't it be better for performance reasons to conditionally exclude that call instead?

Reply:
Nah the JIT should optimize it out

Reply:
It should have, but ... does it and at what point (interpreter)?

Reply:
Taking a closer look at the call site, it seems that  a (cached) reference to the method is passing as an argument to another method, and it's not clear to me how that could be optimized away (but I'm not familiar with the subtleties of JIT compilation), especially since it seems desirable to prevent that other method call altogether.

https://github.com/PowerShell/PowerShell/blob/623c45174ac7c455518f1e0f11fa485110bcb45d/src/System.Management.Automation/engine/runtime/Binding/Binders.cs#L6946-L6958

Reply:
@mklement0 The code you referenced does only one thing - call `LogMemberInvocation` method. This is the code that should have been excluded.

Reply:
Ah, of course, thanks. I agree.

Reply:
The reason why I put the change here is in case the method is called in multiple places in the future. I can certainly change it so the expression tree doesn't call it but I'm not sure if there is any actual overhead to care about with just calling a method that is ultimately a no-op. Will defer to the pwsh team to see what they decide here.

Reply:
Thanks, Jordan. I suppose a pragmatic way to handle this would be to to use `#if UNIX && !DEBUG` around _both_ the currently known only call site as well as inside the method.

Reply:
Updated to follow this pattern.